### PR TITLE
Leave the defaults to broker config - #4694.

### PR DESCRIPTION
**Pull request by alesj**
Signed-off-by: Ales Justin <ales.justin@gmail.com>

### Type of change

- Enhancement / new feature

### Description

We set the replication.factor -- if it's not been set -- to -1, so default broker's settings are used.

### CODE REVIEW

File: topic-operator/src/main/java/io/strimzi/operator/topic/KafkaStreamsTopicStoreService.java [hunk 1]
```diff
@@ -110,8 +111,18 @@ private TopicStore createKafkaTopicStore(Config config, Properties kafkaProperti
             }
         };
 
-        Topology topology = new TopicStoreTopologyProvider(storeTopic, storeName, kafkaProperties, dispatcher).get();
-        streams = new KafkaStreams(topology, kafkaProperties);
+        // provide some Kafka Streams defaults
+        Properties streamsProperties = new Properties();
+        streamsProperties.putAll(kafkaProperties); // don't add them as defaults?
+        Object rf = kafkaProperties.get(StreamsConfig.REPLICATION_FACTOR_CONFIG);
```
Comment (scholzj) [inline topic-operator/src/main/java/io/strimzi/operator/topic/KafkaStreamsTopicStoreService.java:117]: Who and where configures the `kafkaProperties` that we don't know what it contains?

Comment (alesj) [reply topic-operator/src/main/java/io/strimzi/operator/topic/KafkaStreamsTopicStoreService.java:117]: It's configured in Session, out of Config which is setup in Main.

Comment (scholzj) [reply topic-operator/src/main/java/io/strimzi/operator/topic/KafkaStreamsTopicStoreService.java:117]: Thanks 👍

File: topic-operator/src/main/java/io/strimzi/operator/topic/KafkaStreamsTopicStoreService.java [hunk 2]
```diff
@@ -110,8 +111,18 @@ private TopicStore createKafkaTopicStore(Config config, Properties kafkaProperti
             }
         };
 
-        Topology topology = new TopicStoreTopologyProvider(storeTopic, storeName, kafkaProperties, dispatcher).get();
-        streams = new KafkaStreams(topology, kafkaProperties);
+        // provide some Kafka Streams defaults
+        Properties streamsProperties = new Properties();
+        streamsProperties.putAll(kafkaProperties); // don't add them as defaults?
```
Comment (scholzj) [inline topic-operator/src/main/java/io/strimzi/operator/topic/KafkaStreamsTopicStoreService.java:116]: Do we really need the `// don't add them as defaults?` comment? It do not think it adds any value.

**Review by scholzj — APPROVED** [review]
One nit. LGTM otherwise.

**Comment by alesj** [comment]
Yeah, it can be removed ...

**Review by tombentley — APPROVED** [review]

**Pull request merged**